protocol/x/clob/keeper/liquidations.go

```diff
@@ -694,33 +721,56 @@ func (k Keeper) GetPerpetualPositionToLiquidate(
 	// Return an error if there are no perpetual positions to liquidate.
 	if perpetualPosition == nil {
 		return 0,
-			nil,
 			errorsmod.Wrapf(
 				types.ErrNoPerpetualPositionsToLiquidate,
 				"Subaccount ID: %v",
 				subaccount.Id,
 			)
 	}
+	return perpetualPosition.PerpetualId, nil
```

Comment:
instead of returning here, just return where there is `break` above. Then you can also remove the `if perpetualPosition == nil` conditional